src/View/View.php

```diff
@@ -360,12 +360,55 @@ public function __construct(
      * So this method allows you to manipulate them as required after view instance
      * is constructed.
      *
+     * Helpers can be added using {@link addHelper()} method.
+     *
      * @return void
      */
     public function initialize(): void
     {
     }
 
+    /**
+     * Allows setting helpers from initialize() hook, overwriting any previously set ones.
+     *
+     * @param string $helper
+     * @param array<string, mixed> $config
+     * @return void
+     */
+    protected function setHelper(string $helper, array $config = []): void
```

Comment:
I have similar concerns as expressed in https://github.com/cakephp/cakephp/pull/16120/files#r753744719